Add more custom global role information (#1228)
rancher/rancher-product-docs: versions/v2.15/modules/en/pages/rancher-admin/users/authn-and-authz/manage-role-based-access-control-rbac/global-permissions.adoc

Using this field on <<_configuring_default_global_permissions,default GlobalRoles>> may result in users gaining excessive permissions.
====

[#_globalrole_permissions_in_specific_namespaces]
==== GlobalRole Permissions in Specific Namespaces

GlobalRoles can grant access to resources in specific namespaces through two fields, `namespacedRules` and `inheritedNamespacedRules`. The first, `namespacedRules`, is used to provide access to namespaces within the management cluster.

> For example, the following GlobalRole when assigned to a user would allow that user to read secrets in the namespace `my-secrets-namespace` on the management cluster.

Reviewer: ```suggestion
GlobalRoles grant access to resources in specific namespaces using two fields: `namespacedRules` and `inheritedNamespacedRules`. 

The `namespacedRules` field provides access to namespaces within the management cluster. For example, when assigned to a user, the following `GlobalRole` allows that user to read secrets in the `my-secrets-namespace` namespace on the management cluster:
```

[,yaml]
----
apiVersion: management.cattle.io/v3
kind: GlobalRole
displayName: Secret Reader
metadata:
  name: secret-reader
namespacedRules:
  my-secrets-namespace:
    - apiGroups:
      - ""
      resources:
      - secrets
      verbs:
      - "get"
      - "list"
      - "watch"
----

> The second field, `inheritedNamespacedRules`, works similarly, but is intended for namespaces within all downstream clusters. The following example shows a GlobalRole that gives full access to the `cattle-monitoring-system` namespace on every cluster.

Reviewer: ```suggestion
The `inheritedNamespacedRules` field applies permissions to namespaces across all downstream clusters. The following example shows a `GlobalRole` that grants full access to the `cattle-monitoring-system` namespace on every cluster:
```

[,yaml]
----
apiVersion: management.cattle.io/v3
kind: GlobalRole
displayName: Edit Monitoring User
metadata:
  name: monitoring-role
inheritedNamespacedRules:
  cattle-monitoring-system:
    - apiGroups:
      - "*"
      resources:
      - "*"
      verbs:
      - "*"
----

If a cluster does not have the namespace `cattle-monitoring-system`, then no permissions will be provided. If the namespace is then created later on the cluster, the permissions will automatically be added for the user bound to the above GlobalRole.

> Both fields are maps of whose keys are namespaces and values are an array of PolicyRules. That way, multiple namespaces and rules per namespace can be specified.

Reviewer: ```suggestion
If a cluster does not have the `cattle-monitoring-system` namespace, Rancher grants no permissions. If you later create the namespace on that cluster, Rancher automatically adds permissions for users bound to this `GlobalRole`.

Both fields are maps where keys are namespaces and values are arrays of `PolicyRules`. This structure allows you to specify multiple namespaces and multiple rules per namespace.
```

[NOTE]
====

> This field alone does not provide access to the cluster directly. Users still need to be provided permissions to access the cluster, either via `inheritedClusterRoles` or ClusterRoleTemplateBindings.

Reviewer: ```suggestion
These fields alone do not grant direct cluster access. You must also grant users permissions to access the cluster using `inheritedClusterRoles` or `ClusterRoleTemplateBindings`.
```

====

[#_globalrole_permissions_for_fleet_workspaces]
==== GlobalRole Permissions for Fleet Workspaces

> GlobalRoles can grant access to Fleet workspaces in each downstream cluster with the field `inheritedFleetWorkspacePermissions`. This allows users to deploy Fleet resources in all workspaces except `fleet-local`. The field is made up of two parts: 
> 
> * `resourceRules` which grants rules to Fleet resources
> * `workspaceVerbs` which grants permissions to the cluster-wide `fleetworkspace` resources.
> 
> Example:

Reviewer: ```suggestion
GlobalRoles grant access to Fleet workspaces in downstream clusters using the `inheritedFleetWorkspacePermissions` field. This field allows users to deploy Fleet resources in all workspaces except `fleet-local`. The field consists of two parts:

* `resourceRules`: Grants permissions for Fleet resources.
* `workspaceVerbs`: Grants permissions for cluster-wide `fleetworkspace` resources.

The following example shows how to configure Fleet workspace permissions:
```
